org.eclipse.xtext/src/org/eclipse/xtext/validation/DefaultUniqueNameContext.java

```diff
@@ -0,0 +1,320 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Sebastian Zarnekow and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.validation;
+
+import java.util.Collection;
+import java.util.List;
+
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.EcorePackage;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.xtext.resource.IContainer;
+import org.eclipse.xtext.resource.IEObjectDescription;
+import org.eclipse.xtext.resource.IResourceDescription;
+import org.eclipse.xtext.resource.IResourceDescriptions;
+import org.eclipse.xtext.resource.IResourceDescriptionsProvider;
+import org.eclipse.xtext.resource.IResourceServiceProvider;
+import org.eclipse.xtext.resource.ISelectable;
+import org.eclipse.xtext.resource.impl.AbstractCompoundSelectable;
+import org.eclipse.xtext.scoping.ICaseInsensitivityHelper;
+import org.eclipse.xtext.scoping.impl.CaseInsensitivityHelper;
+import org.eclipse.xtext.util.CancelIndicator;
+import org.eclipse.xtext.validation.INamesAreUniqueValidationHelper.Context;
+
+import com.google.common.collect.Iterables;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+/**
+ * Standard implementation of a {@link INamesAreUniqueValidationHelper.Context}.
+ * 
+ * @author Sebastian Zarnekow - Initial contribution and API
+ * @since 2.22
+ */
+public class DefaultUniqueNameContext implements INamesAreUniqueValidationHelper.Context {
+
+	/**
+	 * Base class for {@link INamesAreUniqueValidationHelper.ContextProvider}.
+	 */
+	public static abstract class BaseContextProvider implements INamesAreUniqueValidationHelper.ContextProvider {
+		@Inject
+		private IResourceServiceProvider.Registry resourceServiceProviderRegistry = IResourceServiceProvider.Registry.INSTANCE;
+		@Inject
+		private ICaseInsensitivityHelper caseInsensitivityHelper = new CaseInsensitivityHelper();
+
+		protected IResourceServiceProvider getResourceServiceProvider(Resource r) {
+			return resourceServiceProviderRegistry.getResourceServiceProvider(r.getURI());
+		}
+
+		protected IResourceDescription getResourceDescription(Resource resource) {
+			IResourceDescription.Manager manager = getResourceDescriptionManager(resource);
+			if (manager != null) {
+				return manager.getResourceDescription(resource);
+			}
+			return null;
+		}
+
+		protected IResourceDescription.Manager getResourceDescriptionManager(Resource resource) {
+			IResourceServiceProvider resourceServiceProvider = getResourceServiceProvider(resource);
+			if (resourceServiceProvider == null)
+				return null;
```

Comment:
I would enclose the block with `{` and `}` even if it contains only one statement.